plugins/modules/aws_glue_connection.py

```diff
@@ -249,29 +266,27 @@ def create_or_update_glue_connection(connection, connection_ec2, module, glue_co
                 # We need to slightly modify the params for an update
                 update_params = copy.deepcopy(params)
                 update_params['Name'] = update_params['ConnectionInput']['Name']
-                connection.update_connection(**update_params)
+                if not module.check_mode:
+                    connection.update_connection(**update_params)
                 changed = True
             except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
                 module.fail_json_aws(e)
     else:
         try:
-            connection.create_connection(**params)
+            if not module.check_mode:
+                connection.create_connection(**params)
             changed = True
         except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
             module.fail_json_aws(e)
 
     # If changed, get the Glue connection again
     if changed:
-        glue_connection = None
-        for i in range(10):
-            glue_connection = _get_glue_connection(connection, module)
-            if glue_connection is not None:
-                break
-            time.sleep(10)
+        glue_connection = _get_glue_connection(connection, module)
```

Comment:
The code you've replaced looks like an attempt at a home-made waiter.  This change is likely to result in the module sporadically returning an empty result when it shouldn't be.

See https://www.ansible.com/blog/getting-started-with-aws-ansible-module-development (AWS module development basics) for more details on how to add a waiter.

Reply:
@tremble I replaced the AWSRetry decorators with your suggestion about jittered backoff - thank you!

Glue module doesn't appear to have any waiters. I am in fact not 100% it is necessary at all but since it was in the original module - I wrote my own.